lib/mixins/dir-mixin.html

```diff
@@ -0,0 +1,130 @@
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="../utils/style-gather.html">
+<link rel="import" href="../utils/resolve-url.html">
+<link rel="import" href="element-mixin.html">
+
+<script>
+  (function() {
+    'use strict';
+
+    const HOST_DIR = /:host\(:dir\((ltr|rtl)\)\)/g;
+    const HOST_DIR_REPLACMENT = ':host([dir="$1"])';
+
+    const EL_DIR = /([\s\w#\.\[\]\*]*):dir\((ltr|rtl)\)/g;
+    const EL_DIR_REPLACMENT = ':host([dir="$2"]) $1';
+
+    /**
+     * @type {!Array<!Polymer_DirMixin>}
+     */
+    const DIR_INSTANCES = [];
+
+    let observed = false;
+
+    /**
+     * @param {!Polymer_DirMixin} instance Instance to set RTL status on
+     */
+    function setRTL(instance) {
+      if (!instance.__origRTLStatus) {
+        const el = /** @type {!HTMLElement} */(instance);
+        if (document.documentElement.getAttribute('dir') === 'rtl') {
+          el.setAttribute('dir', 'rtl');
+        } else {
+          el.removeAttribute('dir');
+        }
+      }
+    }
+
+    /**
+     * Element class mixin that allows elements to use the `:dir` CSS Selector to have
+     * text direction specific styling.
+     *
+     * With this mixin, any stylesheet provided in the template will be transform `:dir` into
+     * `:host([dir])` and sync direction with the page via the element's `dir` attribute.
+     *
+     * Elements can opt out of the global page text direction by setting the `dir` attribute
+     * directly in `ready()`.
+     *
+     * Caveats:
+     * - Automatic left-to-right or right-to-left styling is sync'd with the `<html>` element
+     * - Applications must set `<html dir="ltr">` or `<html dir="rtl">` to sync direction
+     * - Opting out of the global direction styling is permanent
+     *
+     * @mixinFunction
+     * @polymer
+     * @appliesMixin Polymer.ElementMixin
+     * @memberof Polymer
+     */
+    Polymer.DirMixin = Polymer.dedupingMixin((base) => {
+
+      if (!observed) {
+        new MutationObserver(() => {
+          DIR_INSTANCES.forEach(i => setRTL(i));
+        }).observe(document.documentElement, {attributes: true, attributeFilter: ['dir']});
+      }
+
+      /**
+       * @constructor
+       * @extends {base}
+       * @implements {Polymer_ElementMixin}
+       */
+      const elementBase = Polymer.ElementMixin(base);
```

Comment:
What about removing this dependency and requiring users to at least also use PropertyAccessors?